price-feeder/cmd/price-feeder.go

```diff
@@ -82,16 +84,51 @@ func priceFeederCmdHandler(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	oracle := oracle.New()
-
 	ctx, cancel := context.WithCancel(context.Background())
 	g, ctx := errgroup.WithContext(ctx)
 
+	// Set up chain
+
+	timeout, err := strconv.Atoi(cfg.RPC.RPCTimeout)
+	if err != nil {
+		trapSignal(cancel)
```

Comment:
We only need to call `trapSignal` once which we already do at the end of this function.